src/Model/Event.php

```diff
@@ -0,0 +1,959 @@
+		}
+
+		$fields = ['start', 'finish', 'adjust', 'summary', 'desc', 'location', 'nofinish'];
+
+		$conditions = ['uid' => $uid, 'cid' => 0];
+
+		// Does the user who requests happen to be the owner of the events
+		// requested? then show all of your events, otherwise only those that
+		// don't have limitations set in allow_cid and allow_gid.
+		if (local_user() != $uid) {
+			$conditions += ['allow_cid' => '', 'allow_gid' => ''];
+		}
+
+		$events = dba::select('event', $fields, $conditions);
+		if (DBM::is_result($events)) {
+			$return = $events;
+		}
+
+		return $return;
+	}
+
+	/**
+	 *
+	 * @param int $uid The user ID.
+	 * @param string $format Output format (ical/csv).
+	 * @return array With the results:
+	 *	bool 'success' => True if the processing was successful,<br>
+	 *	string 'format' => The output format,<br>
+	 *	string 'extension' => The file extension of the output format,<br>
+	 *	string 'content' => The formatted output content.<br>
+	 *
+	 * @todo Respect authenticated users with events_by_uid().
+	 */
+	public static function exportListByUserId($uid, $format = 'ical')
+	{
+		$process = false;
+
+		$user = dba::selectFirst('user', ['timezone'], ['uid' => $uid]);
+		if (DBM::is_result($user)) {
+			$timezone = $user['timezone'];
+		}
+
+		// Get all events which are owned by a uid (respects permissions).
+		$events = self::getListByUserId($uid);
+
+		// We have the events that are available for the requestor.
+		// Now format the output according to the requested format.
+		$res = self::formatListForExport($events, $format, $timezone);
+
+		// If there are results the precess was successfull.
+		if (!empty($res)) {
+			$process = true;
+		}
+
+		// Get the file extension for the format.
+		switch ($format) {
+			case "ical":
+				$file_ext = "ics";
+				break;
+
+			case "csv":
+				$file_ext = "csv";
+				break;
+
+			default:
+				$file_ext = "";
+		}
+
+		$return = [
+			'success'   => $process,
+			'format'    => $format,
+			'extension' => $file_ext,
+			'content'   => $res,
+		];
+
+		return $return;
+	}
+
+	/**
+	 * @brief Format an item array with event data to HTML.
+	 *
+	 * @param arr $item Array with item and event data.
+	 * @return string HTML output.
+	 */
+	public static function getItemHTML($item) {
+		$same_date = false;
+		$finish    = false;
+
+		// Set the different time formats.
+		$dformat       = L10n::t('l F d, Y \@ g:i A'); // Friday January 18, 2011 @ 8:01 AM.
+		$dformat_short = L10n::t('D g:i A'); // Fri 8:01 AM.
+		$tformat       = L10n::t('g:i A'); // 8:01 AM.
+
+		// Convert the time to different formats.
+		$dtstart_dt = day_translate(
+			$item['event-adjust'] ?
+				DateTimeFormat::local($item['event-start'], $dformat)
+				: DateTimeFormat::utc($item['event-start'], $dformat)
+		);
+		$dtstart_title = DateTimeFormat::utc($item['event-start'], $item['event-adjust'] ? DateTimeFormat::ATOM : 'Y-m-d\TH:i:s');
+		// Format: Jan till Dec.
+		$month_short = day_short_translate(
```

Comment:
You could move the function ```day_short_translate``` to this class as some local function. It is only used here.

Reply:
I'll do it when I move `include/text.php`, thanks for the tip, though.

Reply:
You'll remind me when I'll submit the PR I'm sure.